### PR TITLE
Mitigate prometheus RAM flooding

### PR DESCRIPTION
**Pull request by cbeneke**
**What this PR does / why we need it**:

https://github.com/kubernetes/kubernetes/pull/68530 fixes a bug in kubernetes, which floods the metrics created by the controller-manager. As the fix seems not to be backported to v1.11 this change will drop all rest_* series from the controller-manager when any clusterversion v1.11.x is used.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note monitoring
Version v1.11.0 - 1.11.3 Clusters will no longer gather `rest_*` metrics from the controller-manager due to a [bug in kubernetes](https://github.com/kubernetes/kubernetes/pull/68530)
```

### CODE REVIEW

File: api/pkg/resources/prometheus/configmap.go [hunk 1]
```diff
@@ -117,6 +117,13 @@ scrape_configs:
       names:
       - "{{ $.TemplateData.Cluster.Status.NamespaceName }}"
 
+{{- if semverCompare "~1.11" $.TemplateData.Cluster.Spec.Version }}
```
Comment (mrIncompetent) [inline api/pkg/resources/prometheus/configmap.go]: I think it needs to look the following:
```
{{- if semverCompare "~1.11" .TemplateData.Cluster.Spec.Version }}
```

Comment (cbeneke) [reply api/pkg/resources/prometheus/configmap.go]: Oh interesting, copied from the only different line in the template.. updated!

**Comment by cbeneke** [comment]
Hmm.. it seems I don't have access to the build log. I compared the Cluster.Version call to https://github.com/kubermatic/kubermatic/blob/2157720cb737d4f7286d23ee713e518918397697/api/pkg/resources/cloudconfig/configmap.go#L90 .. updated to be in the TemplateData though, as I suspected the structure is comparable. Would be nice if someone can lighten me up how to correctly compare the version there. :)

**Comment by alvaroaleman** [comment]
I asked upstream if there will be a backport because IMHO there should be one: https://github.com/kubernetes/kubernetes/pull/68530#issuecomment-422719606

**Review by alvaroaleman — DISMISSED** [review]

**Comment by cbeneke** [comment]
Thanks for fixing the PR :) If it gets cherry picked, the workaround should still be rolled out for versions v1.11.0 - v1.11.x (x being the release including the backport) for stability. (Or am I missing something?)

**Comment by alvaroaleman** [comment]
@cbeneke Yep exactly

**Review by kron4eg — DISMISSED** [review]

**Comment by mrIncompetent** [comment]
The cherry pick for upstream has been merged.
We can now limit the affected versions to 1.11.0 - 1.11.3

**Comment by mrIncompetent** [comment]
@alvaroaleman @kron4eg @cbeneke 
I updated the config to only drop those metrics for 1.11.0 - 1.11.3.
PTAL

**Review by alvaroaleman — APPROVED** [review]

**Review by kron4eg — APPROVED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**